src/rgw/rgw_sal_motr.cc

```diff
@@ -2715,6 +3010,15 @@ int MotrAtomicWriter::complete(size_t accounted_size, const std::string& etag,
   obj.meta.encode(bl);
   ldpp_dout(dpp, 20) <<__func__<< ": lid=0x" << std::hex << obj.meta.layout_id
                                                            << dendl;
+          
+  if(!info.versioning_enabled()) // suspended and un-versioned cases
+  {
+    // if bkt-version is suspended/unversioned, then fetch & update previous null
+    // version entry instead of adding new null version entry.
+    int ret_rc;
+    ret_rc = obj.update_null_reference(dpp, ent);
+    ldpp_dout(dpp, 20) <<__func__<< ": update_null_reference rc : " << ret_rc << dendl;
+  }
```

Comment:
Why do we need this separate null-version entry at all in addition to the normal entry which we add below (line 3049)?